src/action.py

```diff
@@ -83,31 +236,47 @@ def generate_body(topic, categories, interest, threshold):
                 raise RuntimeError(f"{category} is not a category of {topic}")
         papers = get_papers(abbr)
         papers = [
-            t for t in papers
-            if bool(set(process_subject_fields(t['subjects'])) & set(categories))]
+            t
+            for t in papers
+            if bool(set(process_subject_fields(t["subjects"])) & set(categories))
+        ]
     else:
         papers = get_papers(abbr)
     if interest:
         relevancy, hallucination = generate_relevance_score(
             papers,
             query={"interest": interest},
             threshold_score=threshold,
-            num_paper_in_prompt=8)
+            num_paper_in_prompt=8,
+        )
         body = "<br><br>".join(
-            [f'Title: <a href="{paper["main_page"]}">{paper["title"]}</a><br>Authors: {paper["authors"]}<br>Score: {paper["Relevancy score"]}<br>Reason: {paper["Reasons for match"]}'
-             for paper in relevancy])
+            [
+                f'Title: <a href="{paper["main_page"]}">{paper["title"]}</a><br>Authors: {paper["authors"]}<br>Score: {paper["Relevancy score"]}<br>Reason: {paper["Reasons for match"]}'
+                for paper in relevancy
+            ]
+        )
         if hallucination:
-            body = "Warning: the model hallucinated some papers. We have tried to remove them, but the scores may not be accurate.<br><br>" + body
+            body = (
+                "Warning: the model hallucinated some papers. We have tried to remove them, but the scores may not be accurate.<br><br>"
+                + body
+            )
     else:
         body = "<br><br>".join(
-            [f'Title: <a href="{paper["main_page"]}">{paper["title"]}</a><br>Authors: {paper["authors"]}'
-             for paper in papers])
+            [
+                f'Title: <a href="{paper["main_page"]}">{paper["title"]}</a><br>Authors: {paper["authors"]}'
+                for paper in papers
+            ]
+        )
     return body
 
 
 if __name__ == "__main__":
+    # Load the .env file.
+    load_dotenv()
```

Comment:
I ran into an issue where having `load_dotenv()` come after the openai imports in `utils.py` was causing it to not receive the key from the `env` file. Adding `openai.api_key = os.environ` after line 284 fixed it for me.

Reply:
done, please check again and LMO if it works now

Reply:
testing it now

Reply:
works